### ISSUE TITLE
Missing One/Zero implementation for vectors

### ISSUE THREAD

**Issue by fkaa** [bjz/cgmath#299]
This existed in `0.3`.

**Comment by crharris** [comment]
For 'zero', you can use the default trait implementation of `Vector::zero()`, and for 'one' you can use `VectorN::from_value(1.0)`, but I agree, this isn't as elegant, I miss having them specific to the type, especially convenient for types when they're inferred.

**Comment by bjz** [comment]
Is this what you mean?

```rust
use cgmath::{Vector3, Vector};

let zero = Vector3::zero();
```

`one` on vectors doesn't really make sense - vectors don't close over multiplication, and don't have a multiplicative identity.

**Comment by crharris** [comment]
Perhaps a `unit_xyz `might be more suitably named as to keep the mathematical purity you're looking for?  I think, like the OP, just looking for convenience methods for defaults instead of writing `VectorX::new(1.0, 1.0, 1.0)`.  Typically, I'll just end up writing my own extension trait for the defaults.

**Comment by bjz** [comment]
Yeah, that would be better. Do you find yourself using a `one` vector that much, and if so what for?

**Comment by fkaa** [comment]
@bjz I'm mainly using it for clamping vectors. Maybe cgmath could get a clamp function similar to the ones in the [GLSL spec](https://www.opengl.org/sdk/docs/man/html/clamp.xhtml)?

**Comment by crharris** [comment]
Come to think of it, I only really ever use it for a scale.

**Comment by mhintz** [comment]
Doesn't `from_value` achieve this?

**Comment by bjz** [comment]
Fixed in #345
